### PR TITLE
SISRP-15960, My Campus and Financial Resources links are not showing for ex-students

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.berkeley.edu/browse/SISRP-15960

Includes:
* a spec that tests only ex-student scenarios
* code cleanup

### CODE REVIEW

**Comment by johncrossman** [comment]
*Note:* This PR restores logic that was displaced by PR #5000

File: app/models/links/my_campus_links.rb [hunk 1]
```diff
@@ -17,67 +16,66 @@ class MyCampusLinks
...
-        @section = {
-          "label" => cat.name,
-          "categories" => get_subsections_for_nav(cat)
-        }
-        navigation.push(@section)
-
+      Links::LinkCategory.where('root_level = ?', true).order('LOWER(name)').each do |category|
+        navigation.push({
+                          'label' => category.name,
+                          'categories' => get_subsections_for_nav(category)
```
Comment (paulkerschen) [inline app/models/links/my_campus_links.rb]: @johncrossman As long as we're fixing up the formatting, here and below, should the line-continuation hashes have only two-space indents?

**Comment by paulkerschen** [comment]
Makes sense, @johncrossman, and thanks for the spec. One q on formatting.

File: app/models/links/my_campus_links.rb (hunk 1 above)
Comment (johncrossman) [inline app/models/links/my_campus_links.rb]: Fixed. Thanks, @paulkerschen

**Pull request merged**